doc/services/llext/config.rst

```diff
@@ -0,0 +1,121 @@
+Configuration
+#############
+
+The following Kconfig options are available for the LLEXT subsystem:
+
+.. _llext_kconf_heap:
+
+Heap size
+----------
+
+The LLEXT subsystem needs a static heap to be allocated for extension related
+data. The following option controls this allocation.
+
+:kconfig:option:`CONFIG_LLEXT_HEAP_SIZE`
+
+        Size of the LLEXT heap in kilobytes.
+
+.. note::
+
+   When :ref:`user mode <usermode_api>` is enabled, the heap size must be
+   large enough to allow the extension sections to be allocated with the
+   alignment required by the architecture.
+
+.. _llext_kconf_type:
+
+ELF object type
+---------------
+
+The LLEXT subsystem supports loading different types of extensions; the type
+can be set by choosing among the following Kconfig options:
```

Comment:
`grep`'ing the tree, these Kconfig options seem to only affect how extensions are generated and not the LLEXT subsystem. Proposal:
```suggestion
The LLEXT subsystem is able to load various ELF file types as extensions. When
building extensions, which type is built can be selected by choosing among the
following Kconfig options:
```

(out-of-scope, but shouldn't this choice be turned into a CMake function argument rather than Kconfig, for this reason?)

Reply:
> these Kconfig options seem to only affect how extensions are generated and not the LLEXT subsystem

This is currently true, however that might be due to the minimal overlap between targets and ELF types. I think properly supporting dynamic loading might benefit from knowing the ELF type in advance, to avoid littering address-related code with "is this a VMA or file offset? check the elf type". That's why I preferred to be a little bit more restrictive than necessary.

Reply:
That seems entirely reasonable to me! Agreed